### DISCUSSION TITLE
[Bug report] moving Page into a hierarchy (assigning a parent page) makes link targets in the WYSIWYG editor invalid

### DISCUSSION THREAD

**Discussion by OttoKerner** [GetPublii/Publii#1644] [Bug Report]
### Operating system

Windows 11 23H2

### Publii version

0.46.1 (build 16918)

### Issue type

Application

### Bug description

Pages which are moved into a hierarchy (by assigning a parent page) after creating them, produces invalid links in the WYSIWYG editor.

### Steps to reproduce

1) Create a page at the top level and save it. Let's call this one "Page1"
2) create a new page which is supposed to serve as the "parent" page of the page in step 1. Let's call this "Page2"
3) move the page from step 1 to be a child of the page from step 2. This is moving the page address from `/page1` to `/page2/page1`
4) edit another page with a picture. Add a link to the picture of the link-type "page", pointing to the title of the page from step 1.
5) upload the changes. You will find that the link on the picture is still pointing to `/page1`, instead of `/page2/page1`

**Comment by dziudek** [comment]
Hi,

Unfortunately I am unable to reproduce this issue - I have tried to reproduce it on preview and after syncing test page - in both cases link to the subpage was correct.

After reopening the website do you still have correct pages hierarchy? Maybe structure has not been saved correctly and then has been wrongly interpreted during website rendering?

**Comment by OttoKerner** [comment]
Yes, I have closed and reopened the Publii app several times.
The page structure is still shown as I sorted it.
But the WYSIWYG editor is still inserting the wrong link URL.

What I have noticed is that the translation for the Page-related items in the UI is not complete (in my case German). Could it have to do with the issue (although I doubt it.)
I've also edited the page's slug under SEO of each involved page and re-saved, to no avail.
Is there a way to force the WYSIWYG editor to recreate whatever resource it uses to determine the page address?

**Reply by dziudek** [reply]
@OttoKerner - when you are in the WYSIWG - how the invalid link looks under source code editor? It  uses `#INTERNAL_LINK#` syntax?

**Reply by OttoKerner** [reply]
Yes, it does.
`<a href="#INTERNAL_LINK#/page/2">testlink</a>`

I wonder if this may be caused by the German Umlaut in the page title of the parent page. It's called "Büro" (although its URL is correctly set to `buero`, thus avoiding the umlaut.)

I've now created a new parent page from scratch, avoid the Umlaut altogether. But after moving all the existing child pages underneath of it, the page links are still created wrong.

**Reply by dziudek** [reply]
@OttoKerner - just to be sure - pretty URLs are enabled under settings for this website?

Also - do you see correct slug on the pages list in Publii? (you can enable it under app settings if it is not visible on your website)

**Reply by OttoKerner** [reply]
Yes, pretty URLs are activated.

Yes, the slugs are correctly shown in the pages list. (i.e. as subfolder of the parent folder)

**Reply by dziudek** [reply]
@OttoKerner - it is extremely weird, because the mechanism of generating slugs on the pages list is very similiar to the mechanism used while generating pages.

Could you attach your `input/config/pages.config.json` file? It contains only pages IDs so nothing important will be revealed and I will be able to check if there are no issues in the structure which can lead to this strange behaviour

**Reply by OttoKerner** [reply]
Sure
```
[
    {
        "id": 48,
        "subpages": []
    },
    {
        "id": 50,
        "subpages": [
            {
                "id": 1,
                "subpages": []
            },
            {
                "id": 2,
                "subpages": []
            },
            {
                "id": 3,
                "subpages": []
            },
            {
                "id": 4,
                "subpages": []
            }
        ]
    },
    {
        "id": 6,
        "subpages": []
    },
    {
        "id": 5,
        "subpages": []
    }
]
```
I have modified some translations in the `editorial-2.lang.json` as well as added some custom CSS. But that's about it.

**Comment by dziudek** [comment]
@OttoKerner - is page with ID = 50 published or draft? :)

**Reply by OttoKerner** [reply]
All pages are published.

**Reply by OttoKerner** [reply]
Page ID 50 is just like a list of links. It is not really intended to be shown directly, because the user is always directed to one of the child pages. 
However, when I call up the page using its URL directly, I get my manually placed links, as well as an automatically created list of all the child pages.
All of these links are invalid, as they don't insert the parent page/folder.
The only place where the links are correct is the menu in the side bar.

**Reply by dziudek** [reply]
@OttoKerner - could you share with me backup of your website? I am unable to reproduce this issue in any way. You can send me a backup or link to file with backup to dziudek@gmail.com

**Reply by OttoKerner** [reply]
download link sent per email
